chef/cookbooks/crowbar/templates/default/crowbar-shutdown-services-before-upgrade.sh.erb

```diff
@@ -19,6 +19,16 @@ log "Executing $BASH_SOURCE"
 
 set -x
 
+mkdir -p $UPGRADEDIR
+
+if [[ -f $UPGRADEDIR/crowbar-shutdown-services-before-upgrade-ok ]] ; then
```

Comment:
I wonder if it is possible for services to somehow come back up after this script has been ran? (for example an external actor such as chef restarting them). If something else failed during the shutdown this file could still be touched. Perhaps we need to set -e and do some catching?

Reply:
1. chef is stopped. If user intervenes manually, he should know how to handle the status file.

2. the file is touched at the very end of the script. It's true that something new could have failed but then we do not report it anyway, so how can one know? Only by going through the logs after some mysterious behavior caused by unreported bug. And once user is already at it, they could see what's happening in the log file...

That said, maybe your idea about `-e` is correct, but that would be general improvement of all the upgrade scripts we're using, not just this one. See https://github.com/crowbar/crowbar-core/tree/master/chef/cookbooks/crowbar/templates/default